ARM64/Aarch64 Support (Raspberry Pi4)

##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
I would love to be able to run awx on a raspberry pi 4. The latest raspberry pi with its stats should be plenty to run the full awx stack, and it would be so useful for people like me who run home labs of just raspberry pis. Ive attempted to run install with `dockerhub_base` commented out in inventory to make the playbooks build the images on the pi 4 but i hit this error:
```
TASK [image_build : Build AWX distribution using container] ******************************************************************************************************************
fatal: [localhost -> localhost]: FAILED! => {"changed": false, "msg": "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1 npm --unsafe-perm --prefix awx/ui ci --no-save awx/ui\n\n> fsevents@1.2.4 install /awx/awx/ui/node_modules/fsevents\n> node install\n\n\n> uglifyjs-webpack-plugin@0.4.6 postinstall /awx/awx/ui/node_modules/uglifyjs-webpack-plugin\n> node lib/post_install.js\n\n\n> puppeteer@1.8.0 install /awx/awx/ui/node_modules/puppeteer\n> node install.js\n\n**INFO** Skipping Chromium download. \"PUPPETEER_SKIP_CHROMIUM_DOWNLOAD\" environment variable was found.\n\n> nunjucks@3.1.3 postinstall /awx/awx/ui/node_modules/nunjucks\n> node postinstall-build.js src\n\n\n> chromedriver@2.40.0 install /awx/awx/ui/node_modules/chromedriver\n> node install.js\n\nOnly Linux 64 bits supported.\nnpm ERR! code ELIFECYCLE\nnpm ERR! errno 1\nnpm ERR! chromedriver@2.40.0 install: `node install.js`\nnpm ERR! Exit status 1\nnpm ERR! \nnpm ERR! Failed at the chromedriver@2.40.0 install script.\nnpm ERR! This is probably not a problem with npm. There is likely additional logging output above.\n\nnpm ERR! A complete log of this run can be found in:\nnpm ERR!     /root/.npm/_logs/2020-05-17T02_23_59_085Z-debug.log\nmake: *** [Makefile:503: awx/ui/.release_deps_built] Error 1\n", "status": 2}

PLAY RECAP *******************************************************************************************************************************************************************
localhost                  : ok=10   changed=2    unreachable=0    failed=1    skipped=3    rescued=0    ignored=0
```

Looks like its failing during the npm install when it installs chromedriver... I'm not sure how to fix this since chromedriver seems to explicitly support x86 only.

https://github.com/giggio/node-chromedriver/issues/156

Maybe theres a way to skip chromedriver? I think its used by puppeteer and i assume puppeteer isnt needed during the build? Open to ideas! And I dont mind doing the work!

--- a/awx/ui/package.json
+++ b/awx/ui/package.json
@@ -43,7 +43,7 @@
     "babel-loader": "^7.1.2",
     "babel-plugin-istanbul": "^4.1.5",
     "babel-preset-env": "^1.6.0",
-    "chromedriver": "^2.35.0",
+    "chromedriver": "^77.0.0",
     "clean-webpack-plugin": "^0.1.16",
     "copy-webpack-plugin": "^4.0.1",
     "css-loader": "^0.28.5",

--- a/installer/roles/image_build/defaults/main.yml
+++ b/installer/roles/image_build/defaults/main.yml
@@ -1,3 +1,7 @@
 ---
 create_preload_data: true
 build_dev: false
+
+# Helper vars to construct the proper download URL for the current architecture
+tini_architecture: '{{ { "x86_64": "amd64", "aarch64": "arm64", "armv7": "arm" }[ansible_facts.architecture] }}'
+kubectl_architecture: '{{ { "x86_64": "amd64", "aarch64": "arm64", "armv7": "arm" }[ansible_facts.architecture] }}'

--- a/installer/roles/image_build/templates/Dockerfile.j2
+++ b/installer/roles/image_build/templates/Dockerfile.j2
@@ -35,7 +35,6 @@ RUN dnf -y update && \
     glibc-langpack-en \
     libcurl-devel \
     libffi-devel \
-    libstdc++.so.6 \
     libtool-ltdl-devel \
     make \
     nodejs \
@@ -115,7 +114,6 @@ RUN dnf -y install \
 
 # Install runtime requirements
 RUN dnf -y update && \
-    dnf -y install https://github.com/krallin/tini/releases/download/v0.18.0/tini_0.18.0.rpm && \
     dnf -y install epel-release 'dnf-command(config-manager)' && \
     dnf module -y enable 'postgresql:10' && \
     dnf config-manager --set-enabled PowerTools && \
@@ -140,11 +138,16 @@ RUN dnf -y update && \
     vim-minimal \
     which \
     xmlsec1-openssl && \
-    dnf -y --repofrompath gcloud,https://packages.cloud.google.com/yum/repos/cloud-sdk-el8-x86_64 \
-       --setopt gcloud.gpgkey=https://packages.cloud.google.com/yum/doc/rpm-package-key.gpg \
-      install kubectl && \
     dnf -y install centos-release-stream && dnf -y install "rsyslog >= 8.1911.0" && dnf -y remove centos-release-stream && \
     dnf -y clean all
+
+# Install kubectl
+RUN curl -L -o /usr/bin/kubectl https://storage.googleapis.com/kubernetes-release/release/v1.17.8/bin/linux/{{ kubectl_architecture | default('amd64') }}/kubectl && \
+    chmod a+x /usr/bin/kubectl
+
+# Install tini
+RUN curl -L -o /usr/bin/tini https://github.com/krallin/tini/releases/download/v0.19.0/tini-{{ tini_architecture | default('amd64') }} && \
+    chmod +x /usr/bin/tini
 
 RUN python3 -m ensurepip && pip3 install "virtualenv < 20" supervisor {% if build_dev|bool %}flake8{% endif %}
 
@@ -244,7 +247,7 @@ CMD ["/bin/bash"]
 USER 1000
 EXPOSE 8052
 
-ENTRYPOINT ["tini", "--"]
+ENTRYPOINT ["/usr/bin/tini", "--"]
 CMD /usr/bin/launch_awx.sh
 VOLUME /var/lib/nginx
 {% endif %}

--- a/installer/roles/kubernetes/templates/postgresql-values.yml.j2
+++ b/installer/roles/kubernetes/templates/postgresql-values.yml.j2
@@ -34,12 +34,22 @@ master:
 {{ affinity | to_nice_yaml(indent=2) | indent(width=4, indentfirst=True) }}
 {% endif %}
 {% endif %}
-{% if pg_image_registry is defined %}
 image:
+{% if pg_image_registry is defined %}
   registry: {{ pg_image_registry }}
+{% endif %}
+  # The default bitnami image from the chart doesn't work on ARM
+  repository: postgres
+  tag: '11'
 volumePermissions:
   image:
+{% if pg_image_registry is defined %}
     registry: {{ pg_image_registry }}
+{% endif %}
+    # The default bitnami image from the chart doesn't work on ARM
+    repository: alpine
+    tag: '3'
+{% if pg_image_registry is defined %}
 metrics:
   image:
     registry: {{ pg_image_registry }}

--- a/requirements/requirements_ansible.txt
+++ b/requirements/requirements_ansible.txt
@@ -92,7 +92,7 @@ pycurl==7.43.0.1          # via -r /awx_devel/requirements/requirements_ansible.
 pygments==2.5.2           # via azure-cli-core, knack
 pyjwt==1.7.1              # via adal, azure-cli-core
 pykerberos==1.2.1         # via requests-kerberos
-pynacl==1.3.0             # via paramiko
+pynacl==1.4.0             # via paramiko
 pyopenssl==19.1.0         # via azure-cli-core, requests-credssp
 pyparsing==2.4.5          # via packaging
 python-dateutil==2.8.1    # via adal, azure-storage, botocore, kubernetes

